Sources/NIO/ChannelHandlerError.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+final public class ChannelHandlerError: ChannelInboundHandler {

```

Comment:
This should be called `CloseOnErrorHandler` or something similar.

Reply:
Please also update the file names with the new handler name.

Reply:
Actually @normanmaurer is there prior art in Netty here?

Reply:
We do not include such a handler in Netty, so no :)

Reply:
we also need to prefix this with `NIO`. I.e. `NIOCloseOnErrorHandler` because that's the guarantee we gave in the [public API declaration](https://github.com/apple/swift-nio/blob/master/docs/public-api.md).

The other question is: is this `swift-nio` or `swift-nio-extras`?

Reply:
Good question. I could go either way there.

Reply:
Added the NIO prefix, for the rest just let me know 😀